### ISSUE TITLE
[BUG] Beacon modules tried to be loaded but not specified in beacon configuration file in salt-minion 3005.1 version

### ISSUE THREAD

**Issue by PatrykPachota** [saltstack/salt#63408]
**Description**
After upgrade salt-minion from 3003 to 3005.1 I am facing many errors related to salt beacon modules (that they cannot be loaded, but I do not want them to be loaded and I do not specify them in beacons configuration file). Even if I am trying to disable them updating beacons.conf file like:
  adb:
      - enabled: False
  aix_account:
      - enabled: False
  avahi_announce:
      - enabled: False
  bonjour_announce:
      - enabled: False
  glxinfo:
      - enabled: False
  haproxy:
      - enabled: False
  inotify:
      - enabled: False
  network_settings:
      - enabled: False
     
It is not helping (after every change I restart salt minion). On salt master (3005.1) I do not have any beacons configured. Errors which I see after starting salt-minion:
Jan 04 14:40:48 likeable-verdict-miscellany salt-minion[4351]: [ERROR   ] Unable to load sensehat beacon: sensehat.get_pressure is missing.
Jan 04 14:40:48 likeable-verdict-miscellany salt-minion[4351]: [ERROR   ] Unable to load sh beacon: strace is missing.
Jan 04 14:40:48 likeable-verdict-miscellany salt-minion[4351]: [ERROR   ] Unable to load imgadm beacon: Only available on SmartOS compute nodes.
Jan 04 14:40:48 likeable-verdict-miscellany salt-minion[4351]: [ERROR   ] Unable to load vmadm beacon: Only available on SmartOS compute nodes.
Jan 04 14:40:52 likeable-verdict-miscellany salt-minion[4351]: [ERROR   ] Unable to load aix_account beacon: Only available on AIX systems.
Jan 04 14:40:52 likeable-verdict-miscellany salt-minion[4351]: [ERROR   ] Unable to load avahi_announce beacon: The 'python-avahi' dependency is missing.
Jan 04 14:40:52 likeable-verdict-miscellany salt-minion[4351]: [ERROR   ] Unable to load bonjour_announce beacon: pybonjour library is missing.
Jan 04 14:40:52 likeable-verdict-miscellany salt-minion[4351]: [ERROR   ] Unable to load haproxy beacon: haproxy.get_sessions is missing.

**Steps to Reproduce the behavior**
Installing 3005.1 salt-minion on CentOS8 machine.

**Expected behavior**
I would like to do not have errors in logs when starting salt-minion

**Versions Report**
Salt Version:
          Salt: 3005.1
 
Dependency Versions:
          cffi: 1.15.1
      cherrypy: Not Installed
      dateutil: 2.6.1
     docker-py: 5.0.3
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 2.10.1
       libgit2: Not Installed
      M2Crypto: 0.35.2
          Mako: Not Installed
       msgpack: 0.6.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: 2.21
      pycrypto: Not Installed
  pycryptodome: Not Installed
        pygit2: Not Installed
        Python: 3.6.8 (default, Nov  2 2021, 13:01:57)
  python-gnupg: Not Installed
        PyYAML: 6.0
         PyZMQ: 20.0.0
         smmap: Not Installed
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.4
 
System Versions:
          dist: centos 8 
        locale: UTF-8
       machine: x86_64
       release: 4.18.0-305.25.1.el8_4.x86_64
        system: Linux
       version: CentOS Linux 8

**Comment by OrangeDog** [comment]
I don't think that config disables anything. Instead it tries to configure all the beacons, but incorrectly.

**Comment by PatrykPachota** [comment]
@OrangeDog So do you have maybe idea how can I 'fix' error messages?

**Comment by matih123** [comment]
@OrangeDog all beacons try to be loaded after adding `- disable_during_state_run: true` option to service beacon. Without this option there are no errors.
Maybe there is some workaround to not check services during state.apply but not cause all beacons to be loaded at the same time? Or do you know when this will be fixed?

**Comment by chateaulav** [comment]
experiencing same issue related to all beacons attempting to load. as @matih123  stated with the `disable_during_state_run` setting, i can confirm that the below beacon loads and associated errors do stop showing when this option is removed from your beacon config.

**beacon config** _(With `disable_during_state_run`  removed)_
```
beacons:
  network_settings:
    - coalesce: True
    - interfaces:
{% for interface in salt['grains.get']('ifwatch') %}
        {{ interface }}:
          ipaddr:
{% endfor %}
    - interval: 60
```

**minion log**
```
root@pxe-dev:~# tail /var/log/salt/minion
2024-01-17 06:43:36,832 [salt.loaded.int.beacons.glxinfo:23  ][ERROR   ][880] Unable to load glxinfo beacon: glxinfo is missing.
2024-01-17 06:43:37,625 [salt.loaded.int.beacons.haproxy:24  ][ERROR   ][880] Unable to load haproxy beacon: haproxy.get_sessions is missing.
2024-01-17 06:43:37,627 [salt.loaded.int.beacons.inotify:48  ][ERROR   ][880] Unable to load inotify beacon: pyinotify library is missing.
2024-01-17 06:43:37,628 [salt.loaded.int.beacons.journald:26  ][ERROR   ][880] Unable to load journald beacon: systemd library is missing.
2024-01-17 06:43:37,633 [salt.loaded.int.beacons.sensehat:26  ][ERROR   ][880] Unable to load sensehat beacon: sensehat.get_pressure is missing.
2024-01-17 06:43:37,644 [salt.loaded.int.beacons.smartos_imgadm:42  ][ERROR   ][880] Unable to load imgadm beacon: Only available on SmartOS compute nodes.
2024-01-17 06:43:37,655 [salt.loaded.int.beacons.smartos_vmadm:42  ][ERROR   ][880] Unable to load vmadm beacon: Only available on SmartOS compute nodes.
2024-01-17 06:43:37,657 [salt.loaded.int.beacons.telegram_bot_msg:30  ][ERROR   ][880] Unable to load telegram_bot_msg beacon: telegram library is missing.
2024-01-17 06:43:37,658 [salt.loaded.int.beacons.twilio_txt_msg:31  ][ERROR   ][880] Unable to load twilio_txt_msg beacon: twilio library is missing.
2024-01-17 06:43:37,659 [salt.loaded.int.beacons.watchdog:77  ][ERROR   ][880] Unable to load watchdog beacon: watchdog library is missing.
```
